cores/esp8266/Esp.h

```diff
@@ -172,6 +172,17 @@ class EspClass {
 #else
         uint32_t getCycleCount();
 #endif
+
+        void enableVM();
+        void setDramHeap();
+        void setIramHeap();
+        void setExternalHeap();
+        void setInternalHeap() {setDramHeap();}; // depricated
```

Comment:
1) At this time it is not allowed. 
> Because of the extended processing time that it takes to process a malloc using the  non32-bit exception handler. I chose to force all new malloc's that occur from an IRQ over to DRAM. If they really want IRAM allocated, they have to do the allocation at INTLEVEL 0.

2) It is designed to work. 
> When I took the changes @earlephilhower had made to `umm_malloc` for multiple heaps, it morphed quite a bit more. There is now a heap context pointer. It can with little overhead, switch to an alternate heap context. Not in this PR, I have an experimental branch where WString does all reallocs with IRAM and when it runs out, a slightly modified umm_realloc will OOM internally and switch over and malloc from the DRAM heap and copy. On return the heap context is still IRAM; however, the allocation returned is in DRAM.